src/libraries/System.Memory/tests/Span/CommonPrefixLength.T.cs

```diff
@@ -0,0 +1,124 @@
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length], new char[length], NonDefaultEqualityComparer<char>.Instance));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(15)]
+        public static void FirstShorterAllEqual_ReturnsFirstLength(int length)
+        {
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length], new char[length + 1]));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length], new char[length + 1], null));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length], new char[length + 1], EqualityComparer<char>.Default));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length], new char[length + 1], NonDefaultEqualityComparer<char>.Instance));
+
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length], new char[length + 1]));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length], new char[length + 1], null));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length], new char[length + 1], EqualityComparer<char>.Default));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length], new char[length + 1], NonDefaultEqualityComparer<char>.Instance));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(15)]
+        public static void SecondShorterAllEqual_ReturnsSecondLength(int length)
+        {
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length + 1], new char[length]));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length + 1], new char[length], null));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length + 1], new char[length], EqualityComparer<char>.Default));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length + 1], new char[length], NonDefaultEqualityComparer<char>.Instance));
+
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length + 1], new char[length]));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length + 1], new char[length], null));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length + 1], new char[length], EqualityComparer<char>.Default));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length + 1], new char[length], NonDefaultEqualityComparer<char>.Instance));
+        }
```

Comment:
these tests could be refactored (to reduce duplication and make them easier to read)

```suggestion
        private static void VerifyCommonPrefixLength<T>(int length1, int length2, int expected, IEqualityComparer<T> customComparer)
        {
            Assert.Equal(expected, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<T>)new T[length1], new T[length2]));
            Assert.Equal(expected, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<T>)new T[length1], new T[length2], null));
            Assert.Equal(expected, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<T>)new T[length1], new T[length2], EqualityComparer<T>.Default));
            Assert.Equal(expected, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<T>)new T[length1], new T[length2], customComparer));

            Assert.Equal(expected, MemoryExtensions.CommonPrefixLength((Span<T>)new T[length1], new T[length2]));
            Assert.Equal(expected, MemoryExtensions.CommonPrefixLength((Span<T>)new T[length1], new T[length2], null));
            Assert.Equal(expected, MemoryExtensions.CommonPrefixLength((Span<T>)new T[length1], new T[length2], EqualityComparer<T>.Default));
            Assert.Equal(expected, MemoryExtensions.CommonPrefixLength((Span<T>)new T[length1], new T[length2], customComparer));
        }

        [Theory]
        [InlineData(0, 0)]
        [InlineData(1, 0)]
        [InlineData(0, 1)]
        [InlineData(2, 0)]
        [InlineData(0, 2)]
        public static void OneOrBothZeroLength_Returns0(int length1, int length2)
            => VerifyCommonPrefixLength<char>(length1, length2, 0, NonDefaultEqualityComparer<char>.Instance);

        [Theory]
        [InlineData(1)]
        [InlineData(2)]
        [InlineData(15)]
        public static void SameLengthAllEqual_ReturnsLength(int length)
            => VerifyCommonPrefixLength<char>(length, length, length, NonDefaultEqualityComparer<char>.Instance);

        [Theory]
        [InlineData(1)]
        [InlineData(2)]
        [InlineData(15)]
        public static void FirstShorterAllEqual_ReturnsFirstLength(int length)
            => VerifyCommonPrefixLength<char>(length, length + 1, length, NonDefaultEqualityComparer<char>.Instance);

        [Theory]
        [InlineData(1)]
        [InlineData(2)]
        [InlineData(15)]
        public static void SecondShorterAllEqual_ReturnsSecondLength(int length)
            => VerifyCommonPrefixLength<char>(length + 1, length, length, NonDefaultEqualityComparer<char>.Instance);

```